arrows/vtk/applets/estimate_depth.h

```diff
@@ -21,16 +20,16 @@ class KWIVER_ALGO_VTK_APPLETS_EXPORT estimate_depth
 {
 public:
   estimate_depth();
-  virtual ~estimate_depth();
+  ~estimate_depth() override;
```

Comment:
I'm not sure this is correct to use `override` here since there is no other method called `~estimate_depth()` in a base class (i.e. It is not overriding another method.) Maybe Matthew W. would know.

Reply:
You raise a good point.  I did a little searching and it seems the use of `override` with destructors is a debated issue.  It seems to be allowable by the compilers in our CI, but whether it makes sense is another issue.  This might be one of those things where we need to pick a KWIVER style and add that to the style guide.  I don't care either way.  @mwoehlke-kitware, what are your thoughts?

Reply:
I don't think we're generally using it. AFAIK, compilers don't complain if its missing, like they do for other overridden virtuals. I suspect our general rule is to write `virtual ~T` instead, even if the base class dtor is also virtual.

There are almost 500 instances of explicitly virtual destructors... while this includes the "root" classes, surely many of them are overrides. OTOH there are *exactly three* instances of a dtor marked `override`. (Curiously enough, one of them is `arrows/qt/applets/pipeline_viewer/PipelineWorker.h`, which I would have thought *I* wrote, though it's possible that @KyleFromKitware, who co-authored the original commit, is responsible for the `override` there.)

On the one hand, I could see an argument being made for preferring `override` to ensure that the base class's dtor is also virtual. OTOH I don't know that we have the ability to tool-enforce that. It may be that we would need to write our own clang-tidy check to do so. (Whereas enforcing `~T() override` → `virtual ~T()` is fairly trivial.)

My main "thought" is that if we are going to bless something, it would be nice to bless something that can be tool-enforced.

Reply:
BTW, AFAIK `override` on a dtor means "the same" as on any other member function: this member must override a virtual member in the base class with the same signature... just that in this case, "same signature" means "the destructor". To wit, it is enforcing that the base class's dtor is `virtual`.

That said, I'm not sure how useful that is unless the base class has other virtual methods, and there is already `-Wnon-virtual-dtor` to warn about classes with virtual methods that do not have a virtual dtor, so the chances of this pattern catching real issues that aren't already caught by `-Wnon-virtual-dtor` is probably not high.

Reply:
I will merge this branch as is and then make a new branch to return to `virtual ~T()` since I've already merged another branch which made this same change and I'd like to change both of them.